msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityApplication.java

```diff
@@ -24,9 +24,15 @@ public class ManagedIdentityApplication extends AbstractApplicationBase implemen
 
     private ManagedIdentityApplication(Builder builder) {
         super(builder);
+        super.tokenCache = sharedTokenCache;
+        super.serviceBundle = new ServiceBundle(
+                builder.executorService,
+                builder.httpClient == null ?
```

Comment:
I think this is still a problem. MSI should perform this check even if the app developer customized their `HTTP Client`. And it should only perform it on Service Fabric.

Reply:
In the latest commits I believe I've added the correct behavior:
- For all applications (public/confidential/managed identity) an `HttpClient` can be set by the app developer, or left as default
- For all flows except Service Fabric, we will use the `HttpClient` set at the app-level
- For Service Fabric, we will not use the app-level `HttpClient`, and will instead use one that has the cert validation logic

Essentially, for only Service Fabric we use a new `DefaultHttpClientManagedIdentity` that has the cert validation logic, and if needed that class can be extended to handle any other bespoke MSI logic. Additionally that class is package-private so the app developer should not need to know about it.